### PR TITLE
Update PR template checklist, make the rest a bit lighter

### PR DESCRIPTION
**Pull request by benhoyt**
This updates the checklist in our pull request template. For one thing,
having "pylibjuju" as item 1 -- something that isn't required for most PRs
-- made it seem like the rest of the checklist wasn't that important.
So drop that requirement.

Add coding style and unit tests items. Mention "what is this test testing"
comments. Order so the most obvious/important are first (kind of?).

[View rendered version here.](https://github.com/benhoyt/juju/blob/update-pr-template/PULL_REQUEST_TEMPLATE.md)

### CODE REVIEW

File: PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -1,26 +1,27 @@
...
-*Replace with a description about why this change is needed, along with a description of what changed.*
+*Why this change is needed and what it does.*
 
 ## Checklist
 
- - [ ] Requires a [pylibjuju](https://github.com/juju/python-libjuju) change
- - [ ] Added [integration tests](https://github.com/juju/juju/tree/develop/tests) for the PR
- - [ ] Added or updated [doc.go](https://discourse.charmhub.io/t/readme-in-packages/451) related to packages changed
- - [ ] Comments answer the question of why design decisions were made
+*Strikethrough if not applicable, e.g., `~Go unit tests, with comments saying what you're testing~`.*
```
Comment (barrettj12) [inline PULL_REQUEST_TEMPLATE.md:5]: Is the "e.g." needed? I think "Strikethrough if not applicable" is plenty clear.

**Review by barrettj12 — APPROVED** [review]

File: PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (benhoyt) [reply PULL_REQUEST_TEMPLATE.md]: Thanks, updated.

**Comment by jnsgruk** [comment]
Not required, but might be interesting to fiddle with the PR template [form fields](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/syntax-for-githubs-form-schema). Means that we can drop the `*replace with foo*` and `*remove this*` type entries and just have fields to fill out?

**Comment by cderici** [comment]
I also find a `Notes & Discussion` section really helpful (see example https://github.com/juju/python-libjuju/pull/692).

**Comment by benhoyt** [comment]
> Not required, but might be interesting to fiddle with the PR template [form fields](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/syntax-for-githubs-form-schema). Means that we can drop the `*replace with foo*` and `*remove this*` type entries and just have fields to fill out?

@jnsgruk, thanks for this, I wasn't aware of these. I'll have a play with that.

**Comment by benhoyt** [comment]
@jnsgruk Unfortunately forms are not supported for pull requests, only issues. From [here](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/syntax-for-issue-forms):

> Issue forms are not supported for pull requests. You can create pull request templates in your repositories for collaborators to use.

**Comment by benhoyt** [comment]
Example usage of this new checklist / PR template: https://github.com/juju/juju/pull/14369

**Review by tlm — APPROVED** [review]
docs.go+++++ :)

**Comment by benhoyt** [comment]
Thanks @wallyworld. We can always iterate on this further as needed!

**Pull request merged**